### PR TITLE
store kbid and runtime key in blob

### PR DESCRIPTION
**Pull request by akhilayesantarao**
- store kbid and runtime key in blob
- store kbid locally after create kb
- increase batch size to 50 ( indexer and skillset)

### CODE REVIEW

File: CustomSkillForDataIngestion/Common/Constants.cs [hunk 1]
```diff
@@ -14,7 +14,9 @@ public class Constants
 
         public const string kbContainerName = "qnamakerdata";
 
-        public const string kbIdBlobName = "kbid";
+        public const string kbIdBlobName = "kbid.txt";
+
```
Comment (bingisbestest) [inline CustomSkillForDataIngestion/Common/Constants.cs:18]: remove the .txt

File: CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs [hunk 2]
```diff
@@ -149,46 +149,63 @@ public static class UploadToQnAMaker
...
-        private static async Task<string> GetKbID(ILogger log)
+        private static async Task<string> GetKbID(QnAMakerClient qnaClient, ExecutionContext context, ILogger log)
         {
-            // Create a BlobServiceClient object which will be used to create a container client
-            BlobServiceClient blobServiceClient = new BlobServiceClient(GetAppSetting("AzureWebJobsStorage"));
             string kbId = string.Empty;
-            BlobContainerClient containerClient = blobServiceClient.GetBlobContainerClient(Constants.kbContainerName);
-            BlobClient blobClient = containerClient.GetBlobClient(Constants.kbIdBlobName);
-            if (await blobClient.ExistsAsync())
+            var path = Path.Join(context.FunctionAppDirectory,"..\\qnamaker");
```
Comment (bingisbestest) [inline CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs:155]: D:\home\data\QnAMaker should be the path

**Review by ghost — CHANGES_REQUESTED** [review]

File: CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs [hunk 3]
```diff
@@ -115,7 +115,7 @@ public static class UploadToQnAMaker
                 Endpoint = $"https://{GetAppSetting("QnAServiceName")}.cognitiveservices.azure.com"
             };
 
-            string kbId = await GetKbID(log);
+            string kbId = await GetKbID(qnaClient, context, log);
             var updateKB = InitUpdateKB();
             var indexDocuments = new List<IndexDocument>();
             foreach (var msg in qnaQueueMessage.Values)
```
Comment (bingisbestest) [inline CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs:121]: You also need to filter out invalid file names

File: CustomSkillForDataIngestion/Common/Constants.cs (hunk 1 above)
Comment (akhilayesantarao) [reply CustomSkillForDataIngestion/Common/Constants.cs:18]: done

File: CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs (hunk 3 above)
Comment (akhilayesantarao) [reply CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs:121]: added check for file extension and file size in upload-to-qna function

File: CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs (hunk 2 above)
Comment (akhilayesantarao) [reply CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs]: done

**Review by ghost — APPROVED** [review]

**Pull request merged**